debezium-core/src/main/java/io/debezium/relational/TableIdTransformerQuoted.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.relational;
+
+/**
+ * Generic implementation of a TrableIdTransformer that accepts either
+ * a quote character to use for the start and end of every identifier
+ * part or two separate identifiers to use for start and end of each
+ * part.
+ *
+ */
+public class TableIdTransformerQuoted implements TableIdTransformer {
+
```

Comment:
The name of this class seems a bit strange. What about `TableIdQuoting`, `TableIdQuotingStrategy`, or something that better signifies what it does?

Also, does it make sense to add several static final instances for `DOUBLE_QUOTED`, `SINGLE_QUOTED`, and `SINGLE_AND_DOUBLE_QUOTED` instances? That would eliminate the need for [`PostgresTableIdTransformer`](https://github.com/debezium/debezium/pull/260/files#diff-a7160d27a9ce61bc7d146dda2cd65992), and they'd be more easily reusable.